### PR TITLE
Mitigate PRISMA-2021-0213 (com.fasterxml.jackson.core_jackson-databind)

### PR DESCRIPTION
**Pull request by achraf-mer**
Mitigate PRISMA-2021-0213 (com.fasterxml.jackson.core_jackson-databind), issue described in: https://github.com/FasterXML/jackson-databind/issues/3328

**Before:**
```
dai-deployment-templates/local-rest-scorer$ ../gradlew dependencyInsight --dependency jackson-databind
Starting a Gradle Daemon (subsequent builds will be faster)

> Configure project :
1.1.11-SNAPSHOT

> Task :local-rest-scorer:dependencyInsight
com.fasterxml.jackson.core:jackson-databind:2.11.4 (selected by rule)
   variant "compile" [
      org.gradle.status              = release (not requested)
      org.gradle.usage               = java-api
      org.gradle.libraryelements     = jar (compatible with: classes)
      org.gradle.category            = library (not requested)

      Requested attributes not found in the selected variant:
         org.gradle.dependency.bundling = external
         org.gradle.jvm.version         = 8
   ]

com.fasterxml.jackson.core:jackson-databind:2.11.4
+--- com.fasterxml.jackson.datatype:jackson-datatype-jdk8:2.11.4
|    \--- org.springframework.boot:spring-boot-starter-json:2.4.2
|         \--- org.springframework.boot:spring-boot-starter-web:2.4.2
|              \--- compileClasspath (requested org.springframework.boot:spring-boot-starter-web)
+--- com.fasterxml.jackson.datatype:jackson-datatype-jsr310:2.11.4
|    \--- org.springframework.boot:spring-boot-starter-json:2.4.2 (*)
+--- com.fasterxml.jackson.module:jackson-module-parameter-names:2.11.4
|    \--- org.springframework.boot:spring-boot-starter-json:2.4.2 (*)
\--- org.springframework.boot:spring-boot-starter-json:2.4.2 (*)

(*) - dependencies omitted (listed previously)

A web-based, searchable dependency report is available by adding the --scan option.

BUILD SUCCESSFUL in 6s
1 actionable task: 1 executed
```

**After:**
```
dai-deployment-templates/local-rest-scorer$ ../gradlew dependencyInsight --dependency jackson-databind

> Configure project :
1.1.11-SNAPSHOT

> Task :local-rest-scorer:dependencyInsight
com.fasterxml.jackson.core:jackson-databind:2.12.6
   variant "apiElements" [
      org.gradle.category            = library
      org.gradle.dependency.bundling = external
      org.gradle.libraryelements     = jar (compatible with: classes)
      org.gradle.usage               = java-api
      org.gradle.status              = release (not requested)

      Requested attributes not found in the selected variant:
         org.gradle.jvm.version         = 8
   ]
   Selection reasons:
      - Selected by rule
      - By constraint

com.fasterxml.jackson.core:jackson-databind:2.12.6
+--- com.fasterxml.jackson:jackson-bom:2.12.6
|    +--- com.fasterxml.jackson.datatype:jackson-datatype-jsr310:2.12.6
|    |    +--- org.springframework.boot:spring-boot-starter-json:2.5.8
|    |    |    \--- org.springframework.boot:spring-boot-starter-web:2.5.8
|    |    |         \--- compileClasspath (requested org.springframework.boot:spring-boot-starter-web)
|    |    \--- com.fasterxml.jackson:jackson-bom:2.12.6 (*)
|    +--- com.fasterxml.jackson.module:jackson-module-parameter-names:2.12.6
|    |    +--- org.springframework.boot:spring-boot-starter-json:2.5.8 (*)
|    |    \--- com.fasterxml.jackson:jackson-bom:2.12.6 (*)
|    +--- com.fasterxml.jackson.core:jackson-annotations:2.12.6
|    |    +--- io.swagger:swagger-models:1.5.20 (requested com.fasterxml.jackson.core:jackson-annotations:2.9.5)
|    |    |    +--- io.springfox:springfox-swagger2:3.0.0
|    |    |    |    \--- io.springfox:springfox-boot-starter:3.0.0
|    |    |    |         \--- compileClasspath
|    |    |    \--- io.springfox:springfox-swagger-common:3.0.0
|    |    |         +--- io.springfox:springfox-oas:3.0.0
|    |    |         |    \--- io.springfox:springfox-boot-starter:3.0.0 (*)
|    |    |         \--- io.springfox:springfox-swagger2:3.0.0 (*)
|    |    +--- io.swagger.core.v3:swagger-models:2.1.2 (requested com.fasterxml.jackson.core:jackson-annotations:2.10.1)
|    |    |    \--- io.springfox:springfox-oas:3.0.0 (*)
|    |    +--- com.fasterxml.jackson.datatype:jackson-datatype-jsr310:2.12.6 (*)
|    |    +--- com.fasterxml.jackson:jackson-bom:2.12.6 (*)
|    |    \--- com.fasterxml.jackson.core:jackson-databind:2.12.6 (*)
|    +--- com.fasterxml.jackson.core:jackson-core:2.12.6
|    |    +--- com.fasterxml.jackson.datatype:jackson-datatype-jsr310:2.12.6 (*)
|    |    +--- com.fasterxml.jackson.module:jackson-module-parameter-names:2.12.6 (*)
|    |    +--- com.fasterxml.jackson:jackson-bom:2.12.6 (*)
|    |    +--- com.fasterxml.jackson.datatype:jackson-datatype-jdk8:2.12.6
|    |    |    +--- org.springframework.boot:spring-boot-starter-json:2.5.8 (*)
|    |    |    \--- com.fasterxml.jackson:jackson-bom:2.12.6 (*)
|    |    \--- com.fasterxml.jackson.core:jackson-databind:2.12.6 (*)
|    +--- com.fasterxml.jackson.datatype:jackson-datatype-jdk8:2.12.6 (*)
|    \--- com.fasterxml.jackson.core:jackson-databind:2.12.6 (*)
+--- com.fasterxml.jackson.datatype:jackson-datatype-jdk8:2.12.6 (*)
+--- com.fasterxml.jackson.datatype:jackson-datatype-jsr310:2.12.6 (*)
+--- com.fasterxml.jackson.module:jackson-module-parameter-names:2.12.6 (*)
\--- org.springframework.boot:spring-boot-starter-json:2.5.8 (*)

(*) - dependencies omitted (listed previously)

A web-based, searchable dependency report is available by adding the --scan option.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 7.0.
Use '--warning-mode all' to show the individual deprecation warnings.
See https://docs.gradle.org/6.8.2/userguide/command_line_interface.html#sec:command_line_warnings

BUILD SUCCESSFUL in 910ms
1 actionable task: 1 executed
```

### CODE REVIEW

**Review by nkpng2k — COMMENTED** [review]
LGTM. qq - is there a replacement for scala styling? Generally its not that important, but would be nice to have if there is one.

File: common/jdbc/build.gradle [hunk 1]
```diff
@@ -36,13 +36,3 @@ bootJar {
...
-
-apply plugin: 'scalaStyle'
-
-scalaStyle {
-    //configLocation = "$rootDir/gradle/conf/scalastyle_conf.xml"
-    configLocation = "${rootDir}/conf/scalastyle-config.xml"
-    includeTestSourceDirectory = true
-    source = "src/main/scala"
-    testSource = "src/test/scala"
-}
```
Comment (this) [inline common/jdbc/build.gradle:48]: Any reason to remove scala style plugin. I feel like it is better to have it, especially for future fixes.

File: common/jdbc/build.gradle [hunk 2]
```diff
@@ -9,15 +9,15 @@ dependencies {
...
     implementation group: 'ai.h2o', name: 'mojo2-runtime-impl'
-    implementation group: 'ai.h2o', name: 'sparkling-water-scoring_2.11'
+    implementation group: 'ai.h2o', name: 'sparkling-water-scoring_2.12'
     implementation group: 'org.scala-lang', name: 'scala-library'
-    implementation group: 'org.apache.spark', name: 'spark-core_2.11'
-    implementation group: 'org.apache.spark', name: 'spark-sql_2.11'
-    implementation group: 'org.apache.spark', name: 'spark-mllib_2.11'
+    implementation group: 'org.apache.spark', name: 'spark-core_2.12'
+    implementation group: 'org.apache.spark', name: 'spark-sql_2.12'
+    implementation group: 'org.apache.spark', name: 'spark-mllib_2.12'
```
Comment (this) [inline common/jdbc/build.gradle:16]: Hope that versions we have declared at `gradle/mixins/dependencies.gradle` gets applied here correctly.

**Review by this — APPROVED** [review]
Have a small comment, otherwise LGTM

File: common/jdbc/build.gradle (hunk 1 above)
Comment (achraf-mer) [reply common/jdbc/build.gradle:48]: can't find one compatible with scala 2.12

**Comment by achraf-mer** [comment]
> LGTM. qq - is there a replacement for scala styling? Generally its not that important, but would be nice to have if there is one.

@nkpng2k no package for said plugin for scala 2.12, one could adjust it, but since impact is minor I just removed it.

**Pull request merged**